packages/blade/src/components/AutoComplete/_decisions/decisions.md

```diff
@@ -0,0 +1,240 @@
+  selected={selectedItems}
+  onSelectChange={({ values }) => setSelectedItems(values)}
+/>
+
+
+// Somewhere outside in your UI
+<Box>
+  {selectedItems.map((selectedItemName, index) =>
+    <Tag
+      onDismiss={() => removeItem(index)}
+    >
+      {selectedItemName}
+    </Tag>
+  )}
+</Box>;
+```
+
+### With Different Inactive - Active States
+
+| Code                                 | Inactive                                                                                                                   | Active                                                                                                                     |
+| ------------------------------------ | -------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
+| `<AutoComplete rows="1" />`          | <img width="266" alt="image" src="https://github.com/razorpay/blade/assets/30949385/74b849f3-f1b9-4922-b717-70ffc7ffd56e"> | <img width="276" alt="image" src="https://github.com/razorpay/blade/assets/30949385/055bc4f4-fb57-4129-b27a-0f1966c8ad26"> |
+| `<AutoComplete rows="3" />`          | <img width="420" alt="image" src="https://github.com/razorpay/blade/assets/30949385/ed71318c-889e-407e-86bf-fe4f5902dedf"> | <img width="416" alt="image" src="https://github.com/razorpay/blade/assets/30949385/8a636a3c-2afe-43a2-b283-c5ef40c14362"> |
+| `<AutoComplete rows="expandable" />` | <img width="385" alt="image" src="https://github.com/razorpay/blade/assets/30949385/21b63bc4-804a-4180-953c-119a2b676cd5"> | <img width="394" alt="image" src="https://github.com/razorpay/blade/assets/30949385/1131a1bd-2989-4bfc-b659-207088be3bfb"> |
+
+## Accessibility
+
+AutoComplete will contiue to follow [same keyboard navigations as SelectInput](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Dropdown/_decisions/decisions.md#accessibility).
+
+## Referrences
+
+- [Material UI - AutoComplete](https://mui.com/material-ui/react-autocomplete/)
+  - Used for Controlled AutoComplete Inspiration
+  - UI / UX Inspirations
+- [Primer - AutoComplete](https://primer.style/react/Autocomplete/)
+  - Usage of Tags inside AutoComplete Inspiration
+- [Ariakit - Combobox Filtering](https://ariakit.org/examples/combobox-filtering)
+  - Used for Controlled Filtering Inspiration
+- [React Select](https://react-select.com/home)
+  - Used for Accessibility and Keyboard Navigations Inspiration
+
+## Open Questions
+
+- **Q1:** Is everyone sold on `selected` and `onSelectChange` API? It is inconsistent with SelectInput. Might have to change it in SelectInput as well.
+- **Q2:** What to do with BottomSheetHeader (and BaseHeader)? Should we make it slot like Footer? Should we remove title, subtitle, etc?
+- **Q3:** `rows: '1' | '3' | 'expandable'` vs `height: 'single-line' | 'multi-line' | 'expandable'`
```

Comment:
1, 3, expandable is more flexible. in future we can add 2, 5 or more.

Reply:
> 1, 3, expandable is more flexible. in future we can add 2, 5 or more.

I am wondering if such usecase comes up in future, should we add more values here or should we internally define the max-height.

Because if we decide to change max height in future, the `rows:` API will also change. Whereas with `height:` API, we can define what we mean by multiline

Reply:
so right now i can only have 1 | 3 what if i want to have other numbers too? why that restriction?

Reply:
consistency mostly. Places where you can't layout shift UI needs "1" and places where you can shift UI, there we can standardise the number of rows. We settled on 3 there.

Without restriction, I would also be able to say rows="99" and make it infinitely increase